src/services/clientService.ts

```
@@ -1,13 +1,60 @@
+    const staticDir = path.resolve('static');
+    await fs.mkdir(staticDir, { recursive: true });
+    const indexPath = path.resolve(staticDir, 'index.html');
+
+    try {
+
+        const stats = await fs.stat(indexPath).catch(() => null);
+
+        if (stats && (Date.now() - stats.mtimeMs) < 5 * 60 * 1000) {
+            console.log('Cached index.html is fresh; skipping remote fetch');
+        } else {
+            const res = await fetch(process.env.FRONT_END_URL!);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)